airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcEncryption.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mysql
+
+import io.airbyte.cdk.jdbc.SSLCertificateUtils
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.io.IOException
+import java.net.MalformedURLException
+import java.net.URI
+import java.security.KeyStoreException
+import java.security.NoSuchAlgorithmException
+import java.security.cert.CertificateException
+import java.security.spec.InvalidKeySpecException
+import java.util.HashMap
+import org.apache.commons.lang3.RandomStringUtils
+
+private val log = KotlinLogging.logger {}
+
+class MysqlJdbcEncryption(
+    val sslMode: String? = null,
+    val caCertificate: String? = null,
+    val clientCertificate: String? = null,
+    val clientKey: String? = null,
+    val clientKeyPassword: String? = null,
+) {
+
+    val TRUST_KEY_STORE_URL: String = "trustCertificateKeyStoreUrl"
+    val TRUST_KEY_STORE_PASS: String = "trustCertificateKeyStorePassword"
+    val CLIENT_KEY_STORE_URL: String = "clientCertificateKeyStoreUrl"
+    val CLIENT_KEY_STORE_PASS: String = "clientCertificateKeyStorePassword"
+    val CLIENT_KEY_STORE_TYPE: String = "clientCertificateKeyStoreType"
+    val TRUST_KEY_STORE_TYPE: String = "trustCertificateKeyStoreType"
+    val KEY_STORE_TYPE_PKCS12: String = "PKCS12"
+    val SSL_MODE: String = "sslMode"
+
+    private fun getOrGeneratePassword(): String {
+        if (!clientKeyPassword.isNullOrEmpty()) {
+            return clientKeyPassword
+        } else {
+            return RandomStringUtils.randomAlphanumeric(10)
+        }
+    }
+
+    private fun prepareCACertificateKeyStore(): Pair<URI, String>? {
+        // if config available
+        // if has CA cert - make keystore
+        // if has client cert
```

Comment:
this commentary is good but this line (and also L47) would be more readable if you stuck to your `if <condition> - <action>` template; here it seems this could read as `if no CA cert - done` or something like that.